### DISCUSSION TITLE
Guide for idiots?

### DISCUSSION THREAD

**Discussion by hokagay7** [typekcz/beepboop-steam#5] [General]
Hi,
I am a newbie to these kind of stuff and was intrigued.Any chance someone could write an idiots guide on how to install? Where do dependencies need to be installed to? Etc?
Thanks!

**Comment by typekcz** [comment]
Hello,
this project uses npm that manages dependencies for you.
Only thing you have to install is [Node.js](https://nodejs.org/en/) that also contains npm.
Then you can follow guide in [README.md](https://github.com/typekcz/beepboop-steam/blob/master/README.md).
You have to provide configuration (easiest by creating file config.js in projects folder) and then you can run commans shown in installation section.
On Windows you can run these commands in PowerShell. You can open PowerShell by holding shift and right clicking the "beepboop-steam-master" folder and then choosing "Open PowerShell here".

I understand that this can look complicated for someone who is not familiar with this stuff.
If you want you can add me on Steam and I will guide you through it.

**Reply by tEALyONDERmAN** [reply]
Hello, would you be interested in showing a complete noob how to install and debug this? It would be greatly apreciated..

Thank you

**Reply by tEALyONDERmAN** [reply]
probably a bad move sharing on here but here is my friend code for hot water 175188719

**Comment by hokagay7** [comment]
> Hello,
> this project uses npm that manages dependencies for you.
> Only thing you have to install is [Node.js](https://nodejs.org/en/) that also contains npm.
> Then you can follow guide in [README.md](https://github.com/typekcz/beepboop-steam/blob/master/README.md).
> You have to provide configuration (easiest by creating file config.js in projects folder) and then you can run commans shown in installation section.
> On Windows you can run these commands in PowerShell. You can open PowerShell by holding shift and right clicking the "beepboop-steam-master" folder and then choosing "Open PowerShell here".
> 
> I understand that this can look complicated for someone who is not familiar with this stuff.
> If you want you can add me on Steam and I will guide you through it.

i manage to make it run..but when i try to play sound i got this error (only on 'youtube' and 'playurl' command)
![error](https://user-images.githubusercontent.com/68209911/105316278-9d5d7a00-5bfb-11eb-9eaf-37737fbca832.PNG)

did i do anything wrong?

![error npm](https://user-images.githubusercontent.com/68209911/105357019-28a43300-5c2f-11eb-980f-d134b0df6111.PNG)
forgot to include this one too

**Comment by typekcz** [comment]
Yes, that is expected. That's because by default BeepBoop uses open source version of Chrome called Chromium that does not include non-free codecs.

To run BeebBoop on regular Chrome you have to run this command before launching:
`$env:PUPPETEER_EXECUTABLE_PATH="C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"`
This is for PowerShell and you need to have Chrome installed in that path.
In cmd it should be:
`set PUPPETEER_EXECUTABLE_PATH=C:\Program Files (x86)\Google\Chrome\Application\chrome.exe`

I also forgot to mention that if you want to store your own sounds in BeepBoop, you have to set up a PostgreSQL database.
But this is optional and BeepBoop should run fine without it, but with some features not working.

I should probably rewrite installation guide to include this stuff.

**Comment by hokagay7** [comment]
> Yes, that is expected. That's because by default BeepBoop uses open source version of Chrome called Chromium that does not include non-free codecs.
> 
> To run BeebBoop on regular Chrome you have to run this command before launching:
> `$env:PUPPETEER_EXECUTABLE_PATH="C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"`
> This is for PowerShell and you need to have Chrome installed in that path.
> In cmd it should be:
> `set PUPPETEER_EXECUTABLE_PATH=C:\Program Files (x86)\Google\Chrome\Application\chrome.exe`
> 
> I also forgot to mention that if you want to store your own sounds in BeepBoop, you have to set up a PostgreSQL database.
> But this is optional and BeepBoop should run fine without it, but with some features not working.
> 
> I should probably rewrite installation guide to include this stuff.

Dude it works now..but the only problem is, it took long time for the bot to play audio.Nice job actually

**Comment by hokagay7** [comment]
i read the command prompt and it show this
![error cmd](https://user-images.githubusercontent.com/68209911/105372436-9016ae80-5c40-11eb-91c6-89cdd9a05388.PNG)

it took about 12 min to play an audio by search.Roughly 6 min to play url.Thoughts?

**Comment by typekcz** [comment]
That is weird. It should play almost instantly. I tried those URLs on my machine and on my Heroku instance and both played it almost instantly.
I have no idea what could be going on...
I use ytdl-core to pipe data stream to the player so it doesn't download whole youtube video but is streaming it. Even 10 hours long videos start playing immediately.

**Comment by hokagay7** [comment]
> That is weird. It should play almost instantly. I tried those URLs on my machine and on my Heroku instance and both played it almost instantly.
> I have no idea what could be going on...
> I use ytdl-core to pipe data stream to the player so it doesn't download whole youtube video but is streaming it. Even 10 hours long videos start playing immediately.

Dude you are a genius.I reinstall the dependencies,then start it using powershell instead of node.I use node in the first place because of powershell restriction.I managed to make it run and now everything work just fine.TQ for guiding me :)

**Comment by hokagay7** [comment]
for the second time,,i shouldn't closed issue, other people might need help :p

**Comment by typekcz** [comment]
I'm glad you made it work. I still don't understand why that was happening.
Anyway I'm going to convert this to discussion because it's not really an issue.
I also want to rewrite installation guide to be more thorough.

**Comment by hokagay7** [comment]
Hey it's me again. Your bot has been working flawlessly, we always use it. 
Regarding the tts feature, do you have any suggestion for the tts url? I believe google translate [tts](https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=Failed%20to%20load%20resource:%20the%20server%20responded%20with%20a%20status%20of%20404) required an api "failed to load resource: the server responded with a status of 404".

edit: forgot to mention that, google tts wont work as a referral link.
edit2: like this one https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=beep%20boop

**Comment by typekcz** [comment]
Hi, I glad that you find it usefull!

So the TTS is pain in the butt. Especially when there is functional TTS build into Chrome, but there is no way to send the TTS audio into the "fake microphone".

Google translate TTS should work if I managed to find way to have BeepBoop run on Chrome with extra codecs installed, but that's not the case currently.

We have been using [responsivevoice.org](https://responsivevoice.org/). Looks like they have a free plan, I haven't tried that yet. It probably wasn't there or was too limited when I was setting this up.

So I'm using [this URL](https://texttospeech.responsivevoice.org/v1/text:synthesize?lang=en-GB&engine=g1&name=&pitch=0.5&rate=0.5&volume=1&key=ufdknZDi&gender=female&text=Hello,%20I%20am%20BeepBoop%20and%20I%20do%20beep%20and%20boop). It's taken from their front page example. There is one problem and that is the key parameter in URL. You will occasionally have to update it to the one they use on their examples, because they invalidate the key after some time.

**Reply by hokagay7** [reply]
what a mad lad. Thank you so much I really appreciate it
